### PR TITLE
Fix z-index of ckeditor

### LINKED ISSUE #5375: CKEditor styles underneath dropdown inside block
| Q | A
| --- | ---
| Sulu Version | 2.1
| Browser Version | Chrome 83.0.4103.106
# Actual Behavior

Styles get shifted underneath the dropdown below

# Expected Behavior

That the styles get presented above the dropdown, making it possible to select the styles on the bottom

# Steps to Reproduce

Add more styles to the ckeditor, prevent having to much text and add a dropdown below it

# Possible Solutions

```
.ck.ck-editor {
    z-index: 1;
}
```

instead of

```
.ck.ck-editor {
    z-index: 0;
}
```

Schermafbeelding 2020-06-30 om 13 56 03

### PR DESCRIPTION
**Pull request by luca-rath**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #5375
| Related issues/PRs | -
| License | MIT
| Documentation PR | -

#### What's in this PR?

Fix z-index of ckeditor

### CODE REVIEW

File: src/Sulu/Bundle/AdminBundle/Resources/js/containers/CKEditor5/ckeditor5.scss [hunk 1]
```diff
@@ -25,6 +25,9 @@ $textEditorUnpublishedLinkColor: $gold;
         --ck-font-size-base: $textEditorFontSize;
         --ck-icon-size: $textEditorFontSize;
 
+        /* z-index */
+        --ck-z-default: 0;
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/AdminBundle/Resources/js/containers/CKEditor5/ckeditor5.scss:29]: what happens if we set this to `auto` here instead of `0` does that work?

Comment (luca-rath) [reply src/Sulu/Bundle/AdminBundle/Resources/js/containers/CKEditor5/ckeditor5.scss:29]: That doesn't work unfortunately 😬

Comment (alexander-schranz) [reply src/Sulu/Bundle/AdminBundle/Resources/js/containers/CKEditor5/ckeditor5.scss:29]: also when setting `--ck-z-modal` to auto?:

so:

```scss
--ck-z-default: auto;
--ck-z-modal: auto;
```

Comment (alexander-schranz) [reply src/Sulu/Bundle/AdminBundle/Resources/js/containers/CKEditor5/ckeditor5.scss:29]: what is the behaviour with auto?

Comment (luca-rath) [reply src/Sulu/Bundle/AdminBundle/Resources/js/containers/CKEditor5/ckeditor5.scss:29]:
```css
/* z-index */
--ck-z-default: auto;
--ck-z-modal: auto;
--ck-balloon-panel-arrow-z-index: auto;
```

If I do so, it's the same behaviour as before

image

Comment (luca-rath) [reply src/Sulu/Bundle/AdminBundle/Resources/js/containers/CKEditor5/ckeditor5.scss:29]: Since this solution is working and I didn't find an edge case, where it doesn't work, I don't think it's necessary to further improve this. But if you want to investigate more, feel free to take over this pull request 😉

Comment (alexander-schranz) [reply src/Sulu/Bundle/AdminBundle/Resources/js/containers/CKEditor5/ckeditor5.scss:29]: okay for me. can you test also the following case @nnatter posted here https://github.com/sulu/sulu/pull/5138#issuecomment-604490692

Comment (luca-rath) [reply src/Sulu/Bundle/AdminBundle/Resources/js/containers/CKEditor5/ckeditor5.scss:29]: looks good

**Review by alexander-schranz — APPROVED** [review]

File: src/Sulu/Bundle/AdminBundle/Resources/js/containers/CKEditor5/ckeditor5.scss (hunk 1 above)
Comment (alexander-schranz) [reply src/Sulu/Bundle/AdminBundle/Resources/js/containers/CKEditor5/ckeditor5.scss:29]: good from my side. think would be nice if @nnatter could test it, as he knows most about the edge cases with the zindexes.

Comment (luca-rath) [reply src/Sulu/Bundle/AdminBundle/Resources/js/containers/CKEditor5/ckeditor5.scss:29]: yes, would also appreciate that 🙂

**Comment by niklasnatter** [comment]
Have tested this with the following block:
```xml
<block name="blocks" default-type="editor" minOccurs="0">
    <meta>
        <title lang="en">Content</title>
    </meta>
    <types>
        <type name="editor">
            <meta>
                <title lang="en">Text Line</title>
            </meta>
            <properties>
                <property name="text_editor" type="text_editor">
                    <meta>
                        <title lang="en">Text Editor</title>
                    </meta>
                </property>
                <property name="text_line" type="text_line">
                    <meta>
                        <title lang="en">Text Line</title>
                    </meta>
                </property>
            </properties>
        </type>
        <type name="teaser_selection">
            <meta>
                <title lang="en">Teaser Selection</title>
            </meta>
            <properties>
                <property name="text_editor" type="text_editor">
                    <meta>
                        <title lang="en">Text Editor</title>
                    </meta>
                </property>
                <property name="teaser_selection" type="teaser_selection">
                    <meta>
                        <title lang="en">Teaser Selection</title>
                    </meta>
                </property>
            </properties>
        </type>
    </types>
</block>
```

Can confirm that this solves the following 2 issues 🎉 
Screenshot 2021-01-12 at 15 02 57
Screenshot 2021-01-12 at 15 02 36

**Review by niklasnatter — APPROVED** [review]
